java/server/src/org/openqa/selenium/grid/sessionmap/jdbc/JdbcBackedSessionMap.java

```diff
@@ -0,0 +1,184 @@
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.logging.Logger;
+
+
+public class JdbcBackedSessionMap extends SessionMap implements Closeable {
+
+  private static final Json JSON = new Json();
+  private static final Logger LOG = Logger.getLogger(JdbcBackedSessionMap.class.getName());
+  private static final String separator = "@";
+  private final String  tableName;
+  private final Connection connection;
+  private final String sessionIdCol;
+  private final String sessionCapsCol;
+
+
+  public JdbcBackedSessionMap(Tracer tracer, Connection jdbcConnection, String sessionTableName, String sessionIdCol, String sessionCapsCol)  {
+    super(tracer);
+
+    Require.nonNull("JDBC Connection Object", jdbcConnection);
+    Require.nonNull("Table name", sessionTableName);
+    Require.nonNull("Session id column name", sessionIdCol);
+    Require.nonNull("Session caps column name", sessionCapsCol);
+
+    this.connection = jdbcConnection;
+    this.tableName = sessionTableName;
+    this.sessionIdCol = sessionIdCol;
+    this.sessionCapsCol = sessionCapsCol;
+  }
+
+  public static SessionMap create(Config config) {
+    Tracer tracer = new LoggingOptions(config).getTracer();
+    JdbcSessionMapOptions sessionMapOptions = new JdbcSessionMapOptions(config);
+    String tableName = sessionMapOptions.getJdbcTableName();
+    String sessionIdColName = sessionMapOptions.getJdbcSessionIdColName();
+    String sessionCapsColName = sessionMapOptions.getJdbcSessionCapColName();
+    Connection connection;
+
+    try {
+      connection = sessionMapOptions.getJdbcConnection();
+    } catch (SQLException e) {
+      throw new ConfigException(e.toString());
+    }
+
+    return new JdbcBackedSessionMap(tracer, connection, tableName, sessionIdColName, sessionCapsColName);
+  }
+  @Override
+  public boolean add(Session session) {
+    Require.nonNull("Session to add", session);
+
+    try {
+      return insertSessionStatement(session).executeUpdate() >= 1;
+
+    } catch (SQLException e) {
+      throw new JdbcException(e.getMessage());
+    }
+  }
+
+  @Override
+  public Session get(SessionId id) throws NoSuchSessionException {
+    Require.nonNull("Session ID", id);
+
+    URI uri = null;
+    Capabilities caps = null;
+
+    try {
+      ResultSet sessions = readSessionStatement(id).executeQuery();
+
+      while(sessions.next()) {
+        String sessionIdAndURI = sessions.getString(sessionIdCol);
+        String rawUri = sessionIdAndURI.split(separator)[1];
+        String rawCapabilities = sessions.getString(sessionCapsCol);
+
+        caps = rawCapabilities == null ?
+                            new ImmutableCapabilities() :
+                            JSON.toType(rawCapabilities, Capabilities.class);
+
+        try {
+          uri = new URI(rawUri);
+        }
+        catch (URISyntaxException e) {
```

Comment:
Java style is to put the closing brace and the `catch` statement on the same line `} catch (URISyntaxException e) {`